Exception thrown when running grailsw in stock Grails 5.3.5

### Expected Behavior

`grailsw` runs without any problems, e.g. `grailsw -version`.

### Actual Behaviour

```
user@Latitude-7440:~/Downloads/grails-5.3.5/bin/Dummy$ chmod +x ./grailsw
user@Latitude-7440:~/Downloads/grails-5.3.5/bin/Dummy$ ./grailsw
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at grails.init.Start.main(Start.java:111)
Caused by: java.lang.NoClassDefFoundError: io/micronaut/context/env/PropertySourcePropertyResolver
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1022)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:555)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
        at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:594)
        at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:869)
        at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:979)
        at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:967)
        at grails.util.Environment.resolveCurrentEnvironment(Environment.groovy:286)
        at grails.util.Environment.cacheCurrentEnvironment(Environment.groovy:312)
        at grails.util.Environment.getCurrent(Environment.groovy:278)
        at grails.util.Environment.<clinit>(Environment.groovy:127)
        at org.grails.build.parsing.CommandLineParser.<clinit>(CommandLineParser.java:44)
        at org.grails.cli.GrailsCli.<init>(GrailsCli.groovy:109)
        at org.grails.cli.GrailsCli.main(GrailsCli.groovy:157)
        at org.grails.cli.GrailsCli$main.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:139)
        at grails.init.RunCommand.main(RunCommand.groovy:40)
        ... 5 more
Caused by: java.lang.ClassNotFoundException: io.micronaut.context.env.PropertySourcePropertyResolver
        at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:594)
        at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:869)
        at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:979)
        at groovy.lang.GroovyClassLoader.loadClass(GroovyClassLoader.java:967)
        ... 29 more

```

### Steps To Reproduce

1. Download [grails-5.3.5.zip](https://github.com/grails/grails-core/releases/download/v5.3.5/grails-5.3.5.zip) (https://github.com/grails/grails-core/releases/tag/v5.3.5)
2. Unpack
3. `cd bin/`
4. `./grails create-app Dummy`
5. `cd Dummy/`
6. `chmod +x ./grailsw`
7. `./grailsw`

### Environment Information

- Ubuntu 22.04.3 LTS
- openjdk 11.0.21 2023-10-17
OpenJDK Runtime Environment (build 11.0.21+9-post-Ubuntu-0ubuntu122.04)
OpenJDK 64-Bit Server VM (build 11.0.21+9-post-Ubuntu-0ubuntu122.04, mixed mode, sharing)


### Example Application

_No response_

### Version

5.3.5

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,1 +1,1 @@
-projectVersion=3.0.0-SNAPSHOT
+projectVersion=3.1.0.M1

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,1 +1,1 @@
-projectVersion=3.1.0.M1
+projectVersion=3.1.0-SNAPSHOT

--- a/settings.gradle
+++ b/settings.gradle
@@ -1,4 +1,4 @@
 include 'starter'
 include 'wrapper'
 
-project(":wrapper").name = "grails5-wrapper"
+project(":wrapper").name = "grails5_3_4-wrapper"

--- a/starter/build.gradle
+++ b/starter/build.gradle
@@ -1,7 +1,7 @@
 apply plugin: 'java'
 
 jar {
-    archiveName = "grails-wrapper.jar"
+    archiveFileName = "grails-wrapper.jar"
     manifest {
         attributes 'Main-Class': 'grails.init.Start'
     }

--- a/starter/src/main/java/grails/init/Start.java
+++ b/starter/src/main/java/grails/init/Start.java
@@ -21,7 +21,7 @@ import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
 
 public class Start {
 
-    private static final String PROJECT_NAME = "grails5-wrapper";
+    private static final String PROJECT_NAME = "grails5_3_4-wrapper";
     private static final String WRAPPER_PATH = "/org/grails/" + PROJECT_NAME;
     private static final String DEFAULT_GRAILS_CORE_ARTIFACTORY_BASE_URL = "https://repo.grails.org/grails/core";
     private static final File WRAPPER_DIR = new File(System.getProperty("user.home") + "/.grails/wrapper");

--- a/wrapper/build.gradle
+++ b/wrapper/build.gradle
@@ -94,6 +94,7 @@ dependencies {
     implementation "org.codehaus.groovy:groovy:$groovyVersion"
     implementation "org.springframework.boot:spring-boot-cli:$springBootVersion"
     implementation "org.codehaus.groovy:groovy-ant:$groovyVersion"
+    implementation "io.micronaut:micronaut-inject-groovy:$micronautInjectGroovyVersion"
 }
 
 jar {

--- a/wrapper/gradle.properties
+++ b/wrapper/gradle.properties
@@ -1,5 +1,5 @@
-grailsVersion=5.0.0.M1
-grailsGradlePluginVerison=5.0.0.M2
+grailsVersion=5.0.0
+grailsGradlePluginVerison=5.0.0
 groovyVersion=3.0.5
 springBootVersion=2.5.0
 micronautInjectGroovyVersion=3.8.8

--- a/wrapper/gradle.properties
+++ b/wrapper/gradle.properties
@@ -2,3 +2,4 @@ grailsVersion=5.0.0.M1
 grailsGradlePluginVerison=5.0.0.M2
 groovyVersion=3.0.5
 springBootVersion=2.5.0
+micronautInjectGroovyVersion=3.8.8

